packages/contracts/package.json

```diff
@@ -21,15 +21,16 @@
         "build": "ts-node ./scripts/buildVerifiers.ts && yarn compile && yarn abi && hardhat typechain && tsc",
         "abi": "node ./scripts/abi",
         "compile": "hardhat compile",
-        "test": "hardhat test",
+        "test": "hardhat test --no-compile",
```

Comment:
I think users might want to download a `prover` from server and can use it directly
But if we don't put the config in the prover, the user will have one more step to specify the config. (they might not be able to use the config from `@unirep/circuits`)
So if they want to know the config, they can just query it from the prover
Also, they cannot change the config after the circuits are built. So I think we should not let users specify them if they are not matched configs when the circuits are built.

Yeah but if we do the first thing, we should let other configs to be queried from the `defaultProver` but not `@unirep/circuits`

Reply:
> Why do we need this flag? Hardhat automatically compiles only if files have changed. Having no compile enabled is somewhat annoying during development.

It's weird that the [CI will fail](https://app.circleci.com/pipelines/github/Unirep/Unirep/308/workflows/8c513457-36b7-4ae9-bc45-a5e6bf9b1d91/jobs/1974) if I don't use the flag 😥
It works well in my local but not in the circle CI (the verifiers are not built when running `yarn contracts test`)

Reply:
> It's weird that the [CI will fail](https://app.circleci.com/pipelines/github/Unirep/Unirep/308/workflows/8c513457-36b7-4ae9-bc45-a5e6bf9b1d91/jobs/1974) if I don't use the flag 😥

definitely need to fix this

>  So if they want to know the config, they can just query it from the prover

The problem is there is no way to calculate this, it just has to be a variable stored. There is no common way to do so, each prover will have to implement these variables in different ways. I don't think it's worth adding this extra interface complexity, which might lead to confusion.

Reply:
> The problem is there is no way to calculate this, it just has to be a variable stored.

Yes so I think it can be a variable stored in the prover.
So what do you think how we can handle the config if not in the provers?

Reply:
I don't think we need a config associated with the prover. It's up to the operator to properly manage the keys/prover. To me adding a variable to the prover just seems like adding a point of confusion.

We can use the prover for only the vkey and take the other config params the same as before, just a variable.